internal/handlers/pg/msg_createindexes.go

```diff
@@ -59,9 +59,38 @@ func (h *Handler) MsgCreateIndexes(ctx context.Context, msg *wire.OpMsg) (*wire.
 		return nil, err
 	}
 
-	idxArr, err := common.GetRequiredParam[*types.Array](document, "indexes")
-	if err != nil {
-		return nil, err
+	if collection == "" {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrInvalidNamespace,
+			fmt.Sprintf("Invalid namespace specified '%s.'", db),
+			command,
+		)
+	}
+
+	v, _ := document.Get("indexes")
+	if v == nil {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrMissingField,
+			"BSON field 'createIndexes.indexes' is missing but a required field",
+			document.Command(),
+		)
+	}
+
+	idxArr, ok := v.(*types.Array)
+	if !ok {
+		if _, ok = v.(types.NullType); ok {
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrIndexesWrongType,
+				"invalid parameter: expected an object (indexes)",
+				document.Command(),
+			)
+		}
+
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			"BSON field 'createIndexes.indexes' is the wrong type, expected type 'array'",
```

Comment:
We can use the `commonparams.AliasFromType(v)` here to make an identical error message.